python/python/bystro/proteomics/proteomics_cli.py

```diff
@@ -0,0 +1,158 @@
+"""Provide a CLI for proteomics analysis."""
+import argparse
+from pathlib import Path
+from typing import Any, BinaryIO
+
+import requests
+from bystro.api.cli import DEFAULT_DIR, authenticate, login
+from msgspec import json as mjson
+
+# ruff: noqa: T201
+
+UPLOAD_PROTEIN_ENDPOINT = "/api/jobs/upload_protein/"
+HTTP_STATUS_OK = 200
+
+
+def _package_filename(filename: str) -> tuple[str, tuple[str, BinaryIO, str]]:
+    """Wrap filename in a container suitable for upload through the requests library."""
+    filepath = Path(filename)
+    return (
+        "file",
+        (
+            filepath.name,
+            filepath.open("rb"),
+            "application/octet-stream",
+        ),
+    )
+
+
+def upload_proteomics_dataset(args: argparse.Namespace, *, print_result: bool = True) -> dict[str, Any]:
+    """Upload a fragpipe-TMT dataset through the /api/jobs/upload_protein endpoint.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        The arguments passed to the command
+    print_result : bool, optional
+        Whether to print the result of the upload operation, by default True.
+
+    Returns
+    -------
+    dict
+        A json response from the endpoint.
+
+    """
+    state, auth_header = authenticate(args)
+    url = str(state.url / Path(UPLOAD_PROTEIN_ENDPOINT))
+
+    payload = {
+        "job": mjson.encode(
+            {
+                "protein_abundance_file": Path(args.protein_abundance_file).name,
+                "experiment_annotation_file": Path(args.experiment_annotation_file).name,
+                "proteomics_dataset_type": "fragpipe-TMT",  # we currently only support this format
+            }
+        )
+    }
+
+    files_to_upload = [args.protein_abundance_file, args.experiment_annotation_file]
+    files = [_package_filename(filename) for filename in files_to_upload]
+
+    if print_result:
+        print(f"\nCreating jobs for files: {', '.join(files_to_upload)}\n")
+
+    response = requests.post(url, headers=auth_header, data=payload, files=files, timeout=30)
```

Comment:
I think we'll want a longer timeout here, assuming these files can be larger than say 30MB (1MB/s typical broadband upload speed)

Reply:
ok, assuming GB scale data and some grace for <1MB/s connections, changing this to one hour-- sound reasonable?

Reply:
Yeah that sounds fine, and practically, success rates will drop beyond this.

For large documents, maybe we should ask users to pass us s3 paths + credentials? We could add that support in Sprint 2?